lib/google_drive.rb

```diff
@@ -86,70 +84,50 @@ def self.login_with_oauth(client_or_access_token, proxy = nil)
     # +path+ defaults to ENV["HOME"] + "/.ruby_google_drive.token".
     #
     # You can specify your own OAuth +client_id+ and +client_secret+. Otherwise the default one is used.
-    def self.saved_session(path = nil, proxy = nil, client_id = nil, client_secret = nil)
+    def self.saved_session(path = './.config.json', proxy = nil, client_id = nil, client_secret = nil)
+      config = Config.new(path)
+
+      if client_id.nil? || client_secret.nil?
+        config.call
+      else
+        config.client_id = client_id
+        config.client_secret = client_secret
+        config.save
+      end
 
       if proxy
         raise(
             ArgumentError,
-            "Specifying a proxy object is no longer supported. Set ENV[\"http_proxy\"] instead.")
+            'Specifying a proxy object is no longer supported. Set ENV["http_proxy"] instead.')
       end
 
-      if !client_id && !client_secret
-        client_id = "452925651630-egr1f18o96acjjvphpbbd1qlsevkho1d.apps.googleusercontent.com"
-        client_secret = "1U3-Krii5x1oLPrwD5zgn-ry"
-      elsif !client_id || !client_secret
-        raise(ArgumentError, "client_id and client_secret must be both specified or both omitted")
-      end
-
-      path ||= ENV["HOME"] + "/.ruby_google_drive.token"
-      if ::File.exist?(path)
-        lines = ::File.readlines(path)
-        case lines.size
-          when 1
-            token_data = JSON.parse(lines[0].chomp())
-          when 2
-            # Old format.
-            token_data = nil
-          else
-            raise(ArgumentError, "Not a token file: %s" % path)
-        end
-      else
-        token_data = nil
-      end
+      token_data = config.refresh_token
 
       client = Google::APIClient.new(
-          :application_name => "google_drive Ruby library",
-          :application_version => "0.4.0"
+        :application_name => 'google_drive Ruby library',
+        :application_version => '0.4.0'
       )
-      auth = client.authorization
-      auth.client_id = client_id
-      auth.client_secret = client_secret
-      auth.scope = [
-          "https://www.googleapis.com/auth/drive",
-          "https://spreadsheets.google.com/feeds/"
-      ]
-      auth.redirect_uri = "urn:ietf:wg:oauth:2.0:oob"
-
-      if token_data
-
-        auth.refresh_token = token_data["refresh_token"]
-        auth.fetch_access_token!()
 
+      auth = client.authorization
+      auth.client_id = config.client_id
+      auth.client_secret = config.client_secret
+      auth.scope = config.scope ||
+        ['https://www.googleapis.com/auth/drive', 'https://spreadsheets.google.com/feeds/']
+      auth.redirect_uri = 'urn:ietf:wg:oauth:2.0:oob'
+
+      unless token_data.empty?
```

Comment:
How about using nil as the default of refresh_token? Then this can be simply:

    if token_data